File: fapi/src/main/java/com/netcracker/edu/fapi/service/LikeService.java
package com.netcracker.edu.fapi.service;

import com.netcracker.edu.fapi.models.Like;
import com.netcracker.edu.fapi.models.Post;
import com.netcracker.edu.fapi.models.User;

import java.util.List;

public interface LikeService {

    void addLike(Long postId,Long userId);
    List <Like> getLikesByPostId(Long postId);
}

File: frontend/src/app/services/like.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Like} from "../modules/layout/components/models/like";
import {Comment} from "../modules/layout/components/models/comment";

@Injectable()
export class LikeService {

  constructor(private httpClient: HttpClient) {
  }

  saveLike(postId:number,userId:number): Observable<Like> {
    return this.httpClient.post<Like>('/api/like/post/'+postId+"/user/"+userId, Like);
  }

  public getLikesByPostId(postId: number): Observable<Like[]> {
    return this.httpClient.get<Like[]>('/api/like/post/' + postId);
  }
}
